### ISSUE TITLE
Notifications don't open app since SDK45 (Android)

### ISSUE THREAD

**Issue by amagnolo** [expo/expo#17531]
### Summary

I upgraded a managed app from SDK44 to SDK45 and now, when a notification is received while the app is in background, touching the notification in Android doesn’t open the app as it used to with SDK44.
The listener is called and executed, but the app UI is not shown.

This happens both in Expo GO Android and in the native Android app. When the same code is run in Expo GO Apple it correctly opens the app.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

45

### Environment

  expo-env-info 1.0.3 environment info:
    System:
      OS: Linux 5.17 Pop!_OS 22.04 LTS
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 16.14.2 - /usr/bin/node
      npm: 8.5.0 - /usr/bin/npm
    npmPackages:
      expo: ^45.0.0 => 45.0.4 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      eas-cli: 0.52.0
      expo-cli: 5.4.4
    Expo Workflow: managed

### Reproducible demo

This is the listener:

```
Notifications.addNotificationResponseReceivedListener(response => {
console.log(`received notification "${response.notification.request.content.title}"`);
});
```

**Comment by matze10000** [comment]
I can confirm this (my project is still on Expo 0.44). When I install the older version of the Expo Go app (2.23.2) the notifications open the app again.

**Comment by jnoleau** [comment]
Any updates on this issue ? To complete the "To reproduce": 

- Android 12 device with a native managed app built with eas + sdk45
- send a notification with expo notification tool
- The notification is received correctly
- tap on the notification received => the app will not open as expected 

Don't know if it can help but as suggested by the Android doc https://developer.android.com/about/versions/12/behavior-changes-12#identify-notification-trampolines 

`adb shell dumpsys activity service com.android.systemui/.dump.SystemUIAuxiliaryDumpService` identifies the component acting as notification trampolines :

>Starting Intent { act=expo.modules.notifications.NOTIFICATION_EVENT dat=expo-notifications://notifications/0:1653405478578466%dcd1f1b3f9fd7ecd/actions/expo.modules.notifications.actions.DEFAULT cmp=com.babelfrance.loueragile/expo.modules.notifications.service.NotificationsService (has extras) }

**Comment by edjiang** [comment]
Happening for us as well.

**Comment by amagnolo** [comment]
I hope someone from Expo will act swiftly on this bug, as its severity is at blocker level and impacts not only SDK45 projects, but also SDK44 and previous, when tested with Expo Go. 
At the moment, you can't release a SDK45 native Android app in stores as notifications won't work for a sizable share of users.

**Comment by stri8ed** [comment]
Experiencing the same issue. The listener is executed, while the UI is never displayed. This is happening both within Expo Go, as well as a build of the standalone (managed) app. No such issues on IOS.

**Comment by Kudo** [comment]
i'm working on it. both for new expo-notifications packages as well as new expo go build. will update here once the new versions are ready.

**Comment by Kudo** [comment]
i'm sorry that we might take longer time to release the fix. i found a regression issue from my fix. will try to find a solution and anyway, will keep the status updates here.

**Comment by pedropedruzzi** [comment]
Thanks @Kudo. Please keep us posted on the release.

**Comment by edjiang** [comment]
Can we reopen this issue @Kudo because of the regression? Or is there another ticket

**Comment by brentvatne** [comment]
a large part of the team is currently on a team offsite [at app.js conf](https://appjs.co/) - we'll follow up on this when we are back next week

**Comment by edjiang** [comment]
Appreciate it! Have fun at the conference :)

**Comment by rrufus** [comment]
Hello,

Hope the conference was good guys. What is the latest with this issue?

We are facing problems after upgrading to SDK45 and would prefer not to release the next version of our app before it is fixed.

However two partial workarounds were found:
1. You can access the last notification response from https://docs.expo.dev/versions/latest/sdk/notifications/#uselastnotificationresponse-undefined--notificationresponse--null and then route to somewhere in your app from there. But this requires the user to open your app separately to tapping the notification (unless on Android 10, where we found the notification still opened the app but failed to route, so using `useLastNotificationResponse` fully solved it here).
2. If you enable "Appear on top" in the app settings, then the notifications will open as expected (tested on an S21 Ultra with Android 12). However, I don't think this is a permission you can request in expo or highlight from your app if missing, so it would only function as a temporary workaround.

**Comment by Kudo** [comment]
published `expo-notifications@0.15.3` to address the issues. please try to upgrade expo-notifications and let us know if it still not works for you. thanks!

**Comment by amagnolo** [comment]
I confirm that the original issue is now solved. Good job!

**Comment by jnoleau** [comment]
Hi Kudo, thanks for the update, I can confirm too that the app is now opening when clicking a notification but there is another issue : the `addNotificationResponseReceivedListener` or `useLastNotificationResponse` is now always called with the last received notification as a parameter (expected is to be called with the clicked one !)

I don't know if it is a regression or another issue with SDK45 but this issue does not appear on SDK44.

Test : 

Send 3 notifications with 3 different payloads A then B then C

```js
const lastNotificationResponse = Notifications.useLastNotificationResponse()
  React.useEffect(() => {
    if (
      lastNotificationResponse &&
      lastNotificationResponse.actionIdentifier === Notifications.DEFAULT_ACTION_IDENTIFIER
    ) {
      console.log(lastNotificationResponse.notification.request.content)
    }
  }, [lastNotificationResponse])
``` 

Click on A => the log is "C"
Click on B => the log is "C"

(on SDK44 works well , Click on B => the log is "B")

**Comment by Kudo** [comment]
@jnoleau good catch. to fix the trampoline issue, the fix changed much in the way to send notifications, so the issue you reported should be a regression. i'll take a look this week. thanks!

**Comment by jnoleau** [comment]
Hi @Kudo any update about the new issue 🙏 ?

**Comment by Kudo** [comment]
sorry i was busy for next sdk development and didn't get a chance to look it this week. will try to do in these two days.

**Comment by amagnolo** [comment]
> sorry i was busy for next sdk development and didn't get a chance to look it this week. will try to do it these two days.

Nothing personal Kudo, but this order of priorities is one of the biggest hindrances of Expo for the adoption in business environment. You're often too concentrated on delivering timely "best new things" and neglecting making existing stuff work right.
Expo SDK45 was released almost 2 months ago and was completely unusable for anyone employing notifications on Android. It was probably never tested with current Android OS version before release. What's worse, this bug and its regressions (still open) are not even mentioned in the release notes known problems, making anyone upgrading their apps from previous SDK having to find it out the hard way.

**Comment by Kudo** [comment]
@amagnolo i am apologizing for this problem and agreeing that is critical. that's why i am trying to investigate and propose a fix during the weekend. notifications and android intents are much complicated, so we missed some test cases when released sdk 45. especially in android 12 introduced lot of breaking changes, we didn't notice every edge cases for each expo modules. the root cause of `useLastNotificationResponse` issue is actually from multiple pendingIntent and the latest one will overwrite the previous one. 

anyway this is the fix i'm trying to propose: https://gist.github.com/Kudo/b8d6a7e75d951fa6652fe255c2068bd0
if someone has the expertise for android to review my fix or has a chance to test it by patch-package before we landing this fix, in case i missed some edge test cases again, i am super appreciated for that. i will organize a formal pr, merge and release a fix tomorrow. thanks and apologies again.

**Comment by RodolfoGS** [comment]
@Kudo Thanks a lot for your help with this, I'm trying your patch right now and for now it's working perfectly 🎉
I'll update if I see any other issue

**Comment by Kudo** [comment]
i'm glad to the fix work for you, thanks @RodolfoGS!

**Comment by Kudo** [comment]
@jnoleau we just publish `expo-notifications@0.15.4` to address the issue. hopefully this one fixes all the issues. thanks!

**Comment by edjiang** [comment]
Just to confirm, do we need to release a new binary for this to be applied?

**Comment by RodolfoGS** [comment]
@edjiang Yes, you need to release a new binary since the changes was applied in the native code, see the PR (https://github.com/expo/expo/pull/17974/files)

**Comment by pedropedruzzi** [comment]
Hi, my team is using classic builds (`expo build:android`). Our first test with a new build failed. Do we have to wait for some shared binary SDK release to happen before we can re-build and validate? If so, how to track such a release? Thanks in advance.

**Comment by Kudo** [comment]
@pedropedruzzi classic builds take longer time to deploy the expo-notifications update and it should be ready now. could you please try again?

**Comment by pSnehanshu** [comment]
Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:

```
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
```

I found the code that throws this resides here: https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480

Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.

**Comment by GuoXiaoyang** [comment]
> Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:
> 
> ```
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
> ```
> 
> I found the code that throws this resides here:
> 
> https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480
> 
> Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.

Hi.
Have you resolved this issue? I have the sameissue too. Is the device Samsung with Android 11?

**Comment by pSnehanshu** [comment]
@GuoXiaoyang No, I abandoned the project. Yes, Android 11.

**Comment by Renambl** [comment]
> > Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:
> > ```
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
> > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I found the code that throws this resides here:
> > https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480
> > 
> > Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.
> 
> Hi. Have you resolved this issue? I have the sameissue too. Is the device Samsung with Android 11?

Facing similar issues, did this issue get resolved for Samsung galaxy.

**Comment by komal1502** [comment]
> > > Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:
> > > ```
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
> > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I found the code that throws this resides here:
> > > https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480
> > > 
> > > Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.
> > 
> > 
> > Hi. Have you resolved this issue? I have the sameissue too. Is the device Samsung with Android 11?
> 
> Facing similar issues, did this issue get resolved for Samsung galaxy.

Have you found any solution ? Facing the same issue

**Comment by hafizsheetab** [comment]
> Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:
> 
> ```
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
> 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
> ```
> 
> I found the code that throws this resides here:
> 
> https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480
> 
> Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.



> > > > Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:
> > > > ```
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > I found the code that throws this resides here:
> > > > https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480
> > > > 
> > > > Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.
> > > 
> > > 
> > > Hi. Have you resolved this issue? I have the sameissue too. Is the device Samsung with Android 11?
> > 
> > 
> > Facing similar issues, did this issue get resolved for Samsung galaxy.
> 
> Have you found any solution ? Facing the same issue

Any Solution yet?

**Comment by hafizsheetab** [comment]
> > > > Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:
> > > > ```
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
> > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > I found the code that throws this resides here:
> > > > https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480
> > > > 
> > > > Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.
> > > 
> > > 
> > > Hi. Have you resolved this issue? I have the sameissue too. Is the device Samsung with Android 11?
> > 
> > 
> > Facing similar issues, did this issue get resolved for Samsung galaxy.
> 
> Have you found any solution ? Facing the same issue

Please someone solve this. Have been struggling more than a month from now. Found no solutions

**Comment by Harsh-Pratap-Singh1** [comment]
> > > > > Thanks, @Kudo for all the work. I face some problems with a Samsung Galaxy when I click a notification. When I extracted the bug report, here's what I found:
> > > > > ```
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: FATAL EXCEPTION: main
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Process: com.snehanshu.motorsaarthicustomerapp, PID: 27623
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.snehanshu.motorsaarthicustomerapp/expo.modules.notifications.service.NotificationForwarderActivity}: java.lang.IllegalArgumentException: notification and action should not be null
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8663)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: notification and action should not be null
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationsService$Companion.createNotificationResponseBroadcastIntent(NotificationsService.kt:16)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at expo.modules.notifications.service.NotificationForwarderActivity.onCreate(NotificationForwarderActivity.kt:2)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8290)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8270)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
> > > > > 08-02 00:56:23.231 10552 27623 27623 E AndroidRuntime: 	... 12 more
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > I found the code that throws this resides here:
> > > > > https://github.com/expo/expo/blob/41efff607c27f229c3f008ddda5977b9b433d7d6/packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/NotificationsService.kt#L480
> > > > > 
> > > > > Any idea why this may happen? When the app is either background or killed, clicking the notification does nothing. When the app is on foreground, clicking the notification crashes the app. This happens both on Expo Go and standalone apk.
> > > > 
> > > > 
> > > > Hi. Have you resolved this issue? I have the sameissue too. Is the device Samsung with Android 11?
> > > 
> > > 
> > > Facing similar issues, did this issue get resolved for Samsung galaxy.
> > 
> > 
> > Have you found any solution ? Facing the same issue
> 
> Please someone solve this. Have been struggling more than a month from now. Found no solutions

Did you find any solution for this?

**Comment by RodolfoGS** [comment]
@GuoXiaoyang @Renambl @komal1502 @hafizsheetab @Harsh-Pratap-Singh1
This issue is Closed and the crash that you are reporting is not related to the original, I recommend you to create a new Issue with a reproducible example
